assets/js/services/library_builder.js

```diff
@@ -0,0 +1,30 @@
+app.factory("LibraryBuilder", ["Library", "Module", function(Library, Module) {
+
+  var library = function(id) {
+    return Library.get({id: id});
+  };
+
+  var addModule = function(name, libraryId, cb) {
+    var module = new Module({name: name, library: libraryId});
+    module.$save(cb); 
+  };
+
+  var removeModule = function(moduleId) {
+    var module = Module.get({id: moduleId});
```

Comment:
The get request will be asynchronous, so this should not work. Also why not just make a call to remove by giving it an id? Module.remove({id: moduleId})?